sdk/core/azure_core/benches/http_transport_benchmarks.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{
+    credentials::TokenCredential,
+    fmt::SafeDebug,
+    http::{
+        policies::{BearerTokenCredentialPolicy, Policy},
+        ClientMethodOptions, ClientOptions, HttpClient, Method, Pipeline, RawResponse, Request,
+        TransportOptions, Url,
+    },
+    Result,
+};
+use azure_identity::DeveloperToolsCredential;
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::sync::Arc;
+
+#[derive(Clone, SafeDebug)]
+pub struct TestServiceClientOptions {
+    pub client_options: ClientOptions,
+    pub api_version: Option<String>,
+}
+
+impl Default for TestServiceClientOptions {
+    fn default() -> Self {
+        Self {
+            client_options: ClientOptions::default(),
+            api_version: Some("2023-10-01".to_string()),
+        }
+    }
+}
+
+pub struct TestServiceClient {
+    endpoint: Url,
+    api_version: String,
+    pipeline: Pipeline,
+}
+
+#[derive(Default, SafeDebug)]
+pub struct TestServiceClientGetMethodOptions<'a> {
+    pub method_options: ClientMethodOptions<'a>,
+}
+
+impl TestServiceClient {
+    pub fn new(
+        endpoint: &str,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<TestServiceClientOptions>,
+    ) -> Result<Self> {
+        let options = options.unwrap_or_default();
+        let mut endpoint = Url::parse(endpoint)?;
+        if !endpoint.scheme().starts_with("http") {
+            return Err(azure_core::Error::message(
+                azure_core::error::ErrorKind::Other,
+                format!("{endpoint} must use http(s)"),
+            ));
+        }
+        endpoint.set_query(None);
+
+        let auth_policy: Arc<dyn Policy> = Arc::new(BearerTokenCredentialPolicy::new(
+            credential,
+            vec!["https://vault.azure.net/.default"],
```

Comment:
1. Do we really need to auth against the hardcoded endpoint? Is that part of the test?
2. If we do, seems we shouldn't use KV's scope here.

Reply:
We absolutely do not need to auth against anything - but I was trying to get to as close to the customer scenario as I could possibly get for this benchmark, and that meant I needed to authenticate against *something*. And a keyvault scope was literally the first one I saw - I could use a different one :).